libs/asset_type_s3.py

```diff
@@ -0,0 +1,146 @@
+    def __init__(self, name: str, url: str):
+        AssetType.__init__(self, name)
+        self.url = url
+        self.acls = S3Acl()
+
+    def update_grants(self, grants):
+        """
+        Updates the S3 ACL Grants
+        """
+        map_users_uri = {
+            'http://acs.amazonaws.com/groups/global/AllUsers':
+                'all_users_grants',
+            'http://acs.amazonaws.com/groups/global/AuthenticatedUsers':
+                'any_authenticated_users_grants',
+            'http://acs.amazonaws.com/groups/s3/LogDelivery':
+                's3_logs_delivery_grants'
+        }
+        map_permissions = {
+            'READ': 'LIST',
+            'READ_ACP': 'READ',
+            'WRITE': 'WRITE',
+            'WRITE_ACP': 'WRITE',
+        }
+
+        for grant in grants:
+            if 'URI' not in grant['Grantee'] or grant['Grantee']['URI'] not in map_users_uri:
+                continue
+
+            if map_users_uri[grant['Grantee']['URI']] in [
+                'any_authenticated_users_grants',
+                'all_users_grants'
+            ]:
+                self.public = True
+
+            if grant['Permission'] == 'FULL_CONTROL':
+                self.acls.add_right(map_users_uri[grant['Grantee']['URI']], 'LIST')
+                self.acls.add_right(map_users_uri[grant['Grantee']['URI']], 'READ')
+                self.acls.add_right(map_users_uri[grant['Grantee']['URI']], 'WRITE')
+
+            if grant['Permission'] not in map_permissions:
+                continue
+
+            self.acls.add_right(
+                map_users_uri[grant['Grantee']['URI']],
+                map_permissions[grant['Permission']])
+
+    def report(self, report, brief=False):
+        """
+        Add an asset with only relevent informations
+        """
+        if brief:
+            asset_report = self.report_brief()
+        else:
+            asset_report = {
+                'URL': self.url
+            }
+            if self.public:
+                asset_report['PubliclyAccessible'] = True
+            if not self.acls.block_public_acls and not self.security_issues:
+                asset_report['ACL: BlockPublicAcls'] = False
+            if not self.acls.block_public_policy and not self.security_issues:
+                asset_report['ACL: BlockPublicPolicy'] = False
+            if not self.acls.ignore_public_acls and not self.security_issues:
+                asset_report['ACL: IgnorePublicAcls'] = False
+            if not self.acls.restrict_public_buckets and not self.security_issues:
+                asset_report['ACL: RestrictPublicBuckets'] = False
+            if self.acls.all_users_grants and not self.security_issues:
+                asset_report['ACL: All Users'] = self.acls.all_users_grants
+            if self.acls.any_authenticated_users_grants and not self.security_issues:
+                asset_report['ACL: Any Authenticated Users'] = \
+                    self.acls.any_authenticated_users_grants
```

Comment:
As all of theses check are checking `not self.security_isuses`, why not do something like:
```python3
if not self.security_issues:
  ...
```
Furthermore, you are checking bellow `if self.security_issues`, so it will not add any complicated thing to add an `else` condition

Reply:
Thanks